pkg/ingester/limiter.go

```diff
@@ -222,22 +218,37 @@ func (l *Limiter) convertGlobalToLocalLimit(userID string, globalLimit int) int
 		return 0
 	}
 
+	// Global limit is equally distributed among all the active zones.
+	// The portion of global limit related to each zone is then equally distributed
+	// among all the ingesters belonging to that zoneInside of each zone,
+	zoneCount := l.getZonesCount()
+	ingestersInZoneCount := l.getHealthyIngestersInZoneCount(userID)
+	return int((float64(globalLimit) / float64(zoneCount) / float64(ingestersInZoneCount)) * float64(l.replicationFactor))
+}
+
+func (l *Limiter) getShardSize(userID string) int {
+	return l.limits.IngestionTenantShardSize(userID)
+}
+
+func (l *Limiter) getHealthyIngestersInZoneCount(userID string) int {
+	if l.zoneAwarenessEnabled {
+		return util_math.Max(l.ring.HealthyInstancesInZoneCount(), 1)
+	}
+
+	healthyIngestersCount := l.ring.HealthyInstancesCount()
+
 	// If the number of available ingesters is greater than the tenant's shard
 	// size, then we should honor the shard size because series/metadata won't
 	// be written to more ingesters than it.
 	if shardSize := l.getShardSize(userID); shardSize > 0 {
 		// We use Min() to protect from the case the expected shard size is > available ingesters.
-		numIngesters = util_math.Min(numIngesters, util.ShuffleShardExpectedInstances(shardSize, l.getNumZones()))
```

Comment:
I think we need to keep using `ShuffleShardExpectedInstances`. This protects from the case the shard size has been configured to a value not divisible by the number of zones. In this case we round up.